### DISCUSSION TITLE
"UiUiUi" library on top of U8g2 for creating more complex user interfaces

### DISCUSSION THREAD

**Discussion by dirkhillbrecht** [olikraus/u8g2#1972] [Show and tell]
Hello everyone,

I'd like to point you to [the UiUiUi library](https://github.com/dirkhillbrecht/UiUiUi) here on GitHub or in the Arduino library manager. UiUiUi is a library I develop on top of U8g2 to create more complex interfaces with special focus on full-static design (no heap space usage), efficient display updates and event-driven sketches. You might want to have a look or give it a try.

Best regards,
Dirk Hillbrecht

**Comment by olikraus** [comment]
Wow,  Looks very nice. Very detailed documentation.

So as a conclusion it is mostly a visualization library and will focus displaying existing data, correct?

**Reply by dirkhillbrecht** [reply]
At the moment, yes. But it would be nice if the library was extended allow more interactive elements like menus, confirmation dialogs etc. I think it's designed in a way that these extension are possible quite easily. I decided to go public now because the number of bugs seems to have shrunken to a sensible amount and you actually _can_ do useful things with it.

**Reply by olikraus** [reply]
Indeed it looks already very nice and useful. Thanks again for sharing.
